core/src/main/java/jenkins/install/UpgradeWizard.java

```diff
@@ -101,6 +109,38 @@ public boolean isDue() {
 
         return System.currentTimeMillis() > getStateFile().lastModified();
     }
+    
+    /**
+     * Whether to show the upgrade wizard
+     */
+    public boolean isShowUpgradeWizard() {
+        HttpSession session = Stapler.getCurrentRequest().getSession(false);
+        if(session != null) {
+            return Boolean.TRUE.equals(session.getAttribute(SHOW_UPGRADE_WIZARD_FLAG));
```

Comment:
I'm confused by this boolean test: won't this ALWAYS return false when comparing against a string?